openpgp/symmetric/symmetric.go

```diff
@@ -0,0 +1,166 @@
+package symmetric
+
+import (
+	"io"
+	"crypto"
+	"crypto/sha256"
+	"crypto/hmac"
+
+	"github.com/ProtonMail/go-crypto/openpgp/internal/algorithm"
+)
+
+type PublicKeyAEAD struct {
+	Cipher algorithm.CipherFunction
+	BindingHash [32]byte
+	Key []byte
+}
+
+type PrivateKeyAEAD struct {
+	PublicKey PublicKeyAEAD
+	HashSeed [32]byte
+	Key []byte
+}
+
+type PublicKeyHMAC struct {
```

Comment:
I know I'm not normally the one to say that things should be split up etc, but since you already created a folder and separate classes etc, I think it would make sense to have a separate aead.go and hmac.go in this folder for the different classes, to make them easier to find; and perhaps a util.go for the common stuff (or inline it in both places if it's very short).